specs/fulu/p2p-interface.md

```diff
@@ -494,3 +495,29 @@ column discovery.
 | Key   | Value                                                                                                             |
 | ----- | ----------------------------------------------------------------------------------------------------------------- |
 | `cgc` | Custody group count, `uint64` big endian integer with no leading zero bytes (`0` is encoded as empty byte string) |
+
+##### Next fork digest
+
+A new entry is added to the ENR under the key `nfd`, short for _next fork
+digest_. This entry communicates the digest of the next scheduled fork,
+regardless of whether it is a regular or a Blob-Parameters-Only fork.
+
+If no next fork is scheduled, the `nfd` entry contains the default value for the
+type (i.e., the SSZ representation of a zero-filled array).
+
+| Key   | Value                   |
+| :---- | :---------------------- |
+| `nfd` | SSZ Bytes4 `ForkDigest` |
+
+Furthermore, the existing `next_fork_epoch` field under the `eth2` entry MUST be
+set to the epoch of the next fork, whether a regular fork, _or a BPO fork_.
+
+When discovering and interfacing with peers, nodes MUST evaluate `nfd` alongside
+their existing consideration of the `ENRForkID::next_*` fields under the `eth2`
+key, to form a more accurate view of the peer's intended next fork for the
+purposes of sustained peering. A mismatch indicates that the node MUST
+disconnect from such peers at the fork boundary, but not sooner.
+
+Nodes unprepared to follow the Fulu fork will be unaware of `nfd` entries.
+However, their existing comparison of `eth2` entries (concretely
+`next_fork_epoch`) is sufficient to detect upcoming divergence.
```

Comment:
does any client actually implement that check right now? I don't see how that works in practice since if you publish a release that has `FULU_FORK_EPOCH` set and other nodes in the network haven't upgraded yet then the `next_fork_epoch` wouldn't match with those nodes.

Or do clients ignore this check if `next_fork_epoch` is still set to `18446744073709551615` in either their config or the ENR of the peer?